### ISSUE TITLE
Problem lowering code using timedelta arrays

### ISSUE THREAD

**Issue by luk-f-a** [numba/numba#4540]
Certain numba implementations of numpy functions rely on the `type` function. For example: 

https://github.com/numba/numba/blob/60d2bddd0a9ddf84cf900a70e4b7e17bcbe5dc87/numba/targets/arraymath.py#L246

However, `type` does not work with `timedelta64`, which means that for example, `np.sum(array, axis)` does not work with `timedelta64`.

```python
@numba.njit
def foo(x):
    return np.sum(x,0)

foo(np.ones((2,2))) #works
baz(np.ones((2,2)).astype(np.timedelta64)) #fails
```

This comes from the failure to run `type` on `timedelta64`

```python
def foo(x):
    return 1


@numba.extending.overload(foo)
def foo_fc(x):
    zero = getattr(x, 'dtype', x)(0)
    def foo_impl(x):
        a = type(zero)
        return x
    return foo_impl
    
@numba.njit
def bar(x):
    return foo(x)

bar(np.ones(2)) #works
bar(np.ones(2).astype(np.timedelta64)) #fails
```

**Comment by esc** [comment]
@luk-f-a thanks for reporting this! I was able to reproduce this successfully, so I'll label it as a bug for now.

**Comment by luk-f-a** [comment]
thanks @esc this came from the work on #4472. @stuartarchibald suggested to open a new issue rather than trying to address in the same PR

**Comment by esc** [comment]
@luk-f-a so our current hypothesis is that the `type` builtin won't accept the `timedelta64` as input. The fix probably involves fixing that 'overload'.

**Comment by luk-f-a** [comment]
@stuartarchibald suspects the timedelta types are lacking `__class__` attr resolvers. I don't know how to add them, I couldn't find the right `timedelta`template.

**Comment by stuartarchibald** [comment]
@luk-f-a https://github.com/stuartarchibald/numba/commit/33791d3eecd6af17f58661d5398ee22aba783790 will hopefully fix this problem and let work on #4480 continue.
